### ISSUE TITLE
[bug] Invalid setting '13.1' is not a valid 'settings.compiler.version' value.

### ISSUE THREAD

**Issue by CaiJingLong** [conan-io/conan#13849]
### Environment details

* Operating System+version: macOS build for windowss
* Compiler+version: mingw gcc
* Conan version: 2.0.4
* Python version: 3.11

### Steps to reproduce

The profile:

```ini
[settings]
os=Windows
compiler.version=13.1
```

### Logs

```log
ERROR: Invalid setting '13.1' is not a valid 'settings.compiler.version' value.
Possible values are ['4.1', '4.4', '4.5', '4.6', '4.7', '4.8', '4.9', '5', '5.1', '5.2', '5.3', '5.4', '5.5', '6', '6.1', '6.2', '6.3', '6.4', '6.5', '7', '7.1', '7.2', '7.3', '7.4', '7.5', '8', '8.1', '8.2', '8.3', '8.4', '8.5', '9', '9.1', '9.2', '9.3', '9.4', '9.5', '10', '10.1', '10.2', '10.3', '10.4', '11', '11.1', '11.2', '11.3', '12', '12.1', '12.2']
```

```log
$ /usr/local/Cellar/mingw-w64/11.0.0/toolchain-x86_64/bin/x86_64-w64-mingw32-gcc --version                 

x86_64-w64-mingw32-gcc (GCC) 13.1.0
Copyright (C) 2023 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

```
$ brew info  mingw-w64                                                                                                                                                                       
==> mingw-w64: stable 11.0.0 (bottled)
Minimalist GNU for Windows and GCC cross-compilers
https://sourceforge.net/projects/mingw-w64/
/usr/local/Cellar/mingw-w64/11.0.0 (7,862 files, 973.6MB) *
  Poured from bottle using the formulae.brew.sh API on 2023-05-09 at 08:30:55
From: /Volumes/SanDisk_2T/tmps/brew/homebrew-core/Formula/mingw-w64.rb
License: ZPL-2.1
```

So, the gcc need set 11 or 13.1?

**Comment by memsharded** [comment]
This is being added in https://github.com/conan-io/conan/pull/13848, for 1.60 and later 2.0.5

In any case, please recall that not having some settings defined in ``settings.yml`` is not a blocker, as they can be easily added, you can manage your own ``settings.yml`` with ``conan config install``, and even 2.0 adds a ``settings_user.yml`` to do this easier.

**Comment by memsharded** [comment]
Closed by #13848
